File: Api/Dtos/PostDetail.cs
using System;
using System.Collections.Generic;
using Api.Models;

namespace Api.Dtos
{
    public class PostDetail
    {
        public int PostId { get; set; }
        public string Content { get; set; }
        public DateTime TimePost { get; set; }
        public UserToList User { get; set; }
        public GamesToList Game { get; set; }
        public string ImageUrl { get; set; }
        public int LikeCount { get; set; }
        public int CommentCount { get; set; }
        public bool isLike { get; set; }
        public ICollection<CommentDetail> Comments { get; set; }
        public ICollection<LikeDetail> Likes { get; set; }
    }
}

File: Client/src/app/Models/PostDetail.ts
import { User } from './User';
import { Game } from './Game';
import { PostComment } from './PostComment';

export interface PostDetail {
    postId: number,
    content: string,
    timePost: string,
    user: User,
    game: Game,
    imageUrl: string,
    likeCount: number,
    commentCount: number,
    isLike: boolean,
    likes: User[],
    comments: PostComment[],
}